src/app/AlwaysOn.Shared/Services/CosmosDbService.cs

```diff
@@ -582,162 +450,74 @@ public async Task<IEnumerable<ItemComment>> GetCommentsForCatalogItemAsync(Guid
 
         public async Task AddNewCommentAsync(ItemComment comment)
         {
-            var startTime = DateTime.UtcNow;
-            ItemResponse<ItemComment> response = null;
-            CosmosDiagnostics diagnostics = null;
-            var success = false;
-            var conflict = false;
+            var requestOptions = AppInsightsRequestHandler.CreateOptionsWithOperation<ItemRequestOptions>(nameof(AddNewCommentAsync), _dbClient.Endpoint.Host);
```

Comment:
isnt line basically always the same, with the exception of the method name? can this somehow be simplified?

Reply:
Ideas? One possible simplification would be to move this method from App Insights extensions to `CosmosDbService` (as it's the only place where it's used anyway) - therefore having access to _dbClient directly.

Besides that I'm not keen to try and use some reflection magic to populate the method name automatically :) Also it gives you the option to change the operation name to something else.

Reply:
I refactored it, need to test in e2e now :D